packages/web3-eth-contract/test/integration/conract_deploy.test.ts

```diff
@@ -0,0 +1,137 @@
+import { Contract } from '../../src';
+import { sleep, processAsync } from '../shared_fixtures/utils';
+import { greeterByteCode, greeterContractAbi } from '../shared_fixtures/sources/Greeter';
+import { deployRevertAbi, deployRevertByteCode } from '../shared_fixtures/sources/DeployRevert';
+// eslint-disable-next-line
+import { accounts, clientUrl } from '../../../../.github/test.config';
+
+describe('contract', () => {
+	describe('deploy', () => {
+		let contract: Contract<typeof greeterContractAbi>;
+		let deployOptions: Record<string, unknown>;
+		let sendOptions: Record<string, unknown>;
+
+		beforeEach(() => {
+			contract = new Contract(greeterContractAbi, undefined, {
+				provider: clientUrl,
+			});
+
+			deployOptions = {
+				data: greeterByteCode,
+				arguments: ['My Greeting'],
+			};
+
+			sendOptions = { from: accounts[0].address, gas: '1000000' };
+		});
+
+		it('should deploy the contract', async () => {
+			const deployedContract = await contract.deploy(deployOptions).send(sendOptions);
+
+			expect(deployedContract).toBeDefined();
+		});
+
+		it('should deploy the contract if data is provided at initiation', async () => {
+			contract = new Contract(greeterContractAbi, undefined, {
+				provider: 'http://localhost:8545',
+				data: greeterByteCode,
+				from: accounts[0].address,
+				gas: '1000000',
+			});
+			const deployedContract = await contract.deploy({ arguments: ['Hello World'] }).send();
+
+			expect(deployedContract).toBeDefined();
+		});
+
+		it('should return instance of the contract', async () => {
+			const deployedContract = await contract.deploy(deployOptions).send(sendOptions);
+
+			expect(deployedContract).toBeInstanceOf(Contract);
+		});
+
+		it('should set contract address on new contract instance', async () => {
+			const deployedContract = await contract.deploy(deployOptions).send(sendOptions);
+
+			expect(deployedContract.options.address).toBeDefined();
+		});
+
+		// TODO: It works fine but tests hangs because of confirmation handler
+		it.skip('should emit the "confirmation" event', async () => {
+			const confirmationHandler = jest.fn();
+
+			contract
+				.deploy(deployOptions)
+				.send(sendOptions)
+				.on('confirmation', confirmationHandler);
+
+			// Deploy once again to trigger block mining to trigger confirmation
```

Comment:
Does this only happen when testing against Ganache? I believe there's a setting that continuously mines regardless if there's transactions to be included in the block

Reply:
I found it only in the `ganache`. If you are aware of that setting please share.

Reply:
It looks like we'd need to pass `-b 1` which would mine a block every 1 second